Make less known wayland compositors usable

gnidorah opened the pull request:
###### Motivation for this change

They are currently either broken or not usable out of box. This PR fixes that.

way-cooler:
- [x] update to fix build https://github.com/NixOS/nixpkgs/pull/30088
- [x] fix crash
- [x] locker by alt+shift+l

via module
- [x] wc-grab and way-cooler-bg extensions
- [x] terminal by alt+enter
- [x] menu by alt+d
- [x] optional bar

orbment:
- [x] terminal by win+enter
- [x] menu by win+p

velox:
- [x] update to make it work
must be launched via swc-launch https://github.com/michaelforney/velox/issues/25
- [x] terminal by win+shift+enter
- [x] menu by win+r

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

gnidorah commented:
/cc @peterhoeg Thanks!

peterhoeg commented:
Cc: @miltador as way-cooler maintainer.

miltador commented:
LGTM for `way-cooler` changes.

gnidorah commented:
Closing for one more change

gnidorah commented:
I'm also interested in adding way-cooler module to this PR, if its author will provide lemonbar config

FRidh commented on pkgs/applications/window-managers/orbment/default.nix:
```diff
@@ -1,9 +1,15 @@
 
-stdenv.mkDerivation rec {
+let
+  callPackage =  newScope self;
```
why not pass in `callPackage` instead?

gnidorah replied:
@FRidh Fixed, thanks!

gnidorah commented:
Ok, done.

zimbatm commented on pkgs/applications/window-managers/velox/default.nix:
```diff
@@ -1,25 +1,53 @@
+    st-wayland = callPackage ./st.nix {
+      conf = stConf;
+      patches = stPatches;
+    };
```
why not add these to all-packages.nix?

gnidorah replied:
@zimbatm As noted in https://github.com/NixOS/nixpkgs/pull/32285/commits/0520be305a95f70d92b312a7f3a06daf5b51363b these are unusable outside of velox.

Commit:
bemenu: init at 2017-02-14

Commit:
velox: 2015-11-03 -> 2017-07-04

Commit:
orbment, velox: don't expose subprojects

the development of orbment and velox got stuck
their subprojects (bemenu, dmenu-wayland, st-wayland) don't work correctly outside of parent projects
so hide them to not confuse people
swc and wld libraries are unpopular and unlike wlc are not used by anything except velox

Commit:
pythonPackages.pydbus: init at 0.6.0

Commit:
way-cooler: 0.5.2 -> 0.6.2

Commit:
nixos/way-cooler: add module

Commit:
dconf module: use for wayland

non-invasive approach for #31293
see discussion at #32210

Commit:
sway: embed LD_LIBRARY_PATH for #32755

miltador commented:
Is it going to be merged?

Commit:
way-cooler: switch from buildRustPackage to buildRustCrate #31150

zimbatm commented:
Started nox-review on my build machine. If it passes I will merge.

gnidorah commented:
@miltador @zimbatm Thanks!
